java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -113,7 +113,15 @@ private static ArrowType getArrowType(Type type) {
     schema.getObjectProps().forEach((k,v) -> metadata.put(k, v.toString()));
 
     if (type == Type.RECORD) {
-      throw new UnsupportedOperationException();
+      for (Schema.Field field : schema.getFields()) {
+        final ArrowType arrowType = getArrowType(field.schema().getType());
+        if (arrowType != null) {
+          final FieldType fieldType = new FieldType(true, arrowType, null, null);
```

Comment:
please add comments /*parameter name = */ for  "true" and the null values

Reply:
Thanks, fixed.